{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1b8f9fd6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:21:32.964937Z",
     "iopub.status.busy": "2025-09-27T23:21:32.964516Z",
     "iopub.status.idle": "2025-09-27T23:21:34.414133Z",
     "shell.execute_reply": "2025-09-27T23:21:34.412639Z"
    },
    "papermill": {
     "duration": 1.456572,
     "end_time": "2025-09-27T23:21:34.416180",
     "exception": false,
     "start_time": "2025-09-27T23:21:32.959608",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'stylegan3'...\r\n",
      "remote: Enumerating objects: 212, done.\u001b[K\r\n",
      "remote: Counting objects: 100% (163/163), done.\u001b[K\r\n",
      "remote: Compressing objects: 100% (73/73), done.\u001b[K\r\n",
      "remote: Total 212 (delta 99), reused 90 (delta 90), pack-reused 49 (from 1)\u001b[K\r\n",
      "Receiving objects: 100% (212/212), 4.16 MiB | 30.01 MiB/s, done.\r\n",
      "Resolving deltas: 100% (108/108), done.\r\n"
     ]
    }
   ],
   "source": [
    "!git clone https://github.com/NVlabs/stylegan3.git"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "211c37f4",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:21:34.426416Z",
     "iopub.status.busy": "2025-09-27T23:21:34.425530Z",
     "iopub.status.idle": "2025-09-27T23:21:34.433290Z",
     "shell.execute_reply": "2025-09-27T23:21:34.432011Z"
    },
    "papermill": {
     "duration": 0.014911,
     "end_time": "2025-09-27T23:21:34.435244",
     "exception": false,
     "start_time": "2025-09-27T23:21:34.420333",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "当前目录: /kaggle/working/stylegan3\n",
      "['environment.yml', '.github', 'docs', 'train.py', 'dnnlib', 'calc_metrics.py', 'avg_spectra.py', 'Dockerfile', 'LICENSE.txt', 'viz', 'gen_images.py', '.git', 'README.md', 'visualizer.py', 'training', 'legacy.py', 'metrics', 'gui_utils', 'dataset_tool.py', 'gen_video.py', 'torch_utils']\n"
     ]
    }
   ],
   "source": [
    "# 改变当前目录到项目文件夹\n",
    "import os\n",
    "os.chdir('/kaggle/working/stylegan3')\n",
    "\n",
    "# 验证一下当前目录\n",
    "print(\"当前目录:\", os.getcwd())\n",
    "# 列出文件，确认项目文件都在\n",
    "print(os.listdir())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "981f24c9",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2025-09-27T23:21:34.444317Z",
     "iopub.status.busy": "2025-09-27T23:21:34.443916Z",
     "iopub.status.idle": "2025-09-27T23:26:02.950150Z",
     "shell.execute_reply": "2025-09-27T23:26:02.948492Z"
    },
    "jupyter": {
     "outputs_hidden": true
    },
    "papermill": {
     "duration": 268.513753,
     "end_time": "2025-09-27T23:26:02.953035",
     "exception": false,
     "start_time": "2025-09-27T23:21:34.439282",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pillow==8.3.1\r\n",
      "  Downloading Pillow-8.3.1.tar.gz (48.7 MB)\r\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m48.7/48.7 MB\u001b[0m \u001b[31m30.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\r\n",
      "Building wheels for collected packages: pillow\r\n",
      "  Building wheel for pillow (setup.py) ... \u001b[?25l\u001b[?25hdone\r\n",
      "  Created wheel for pillow: filename=Pillow-8.3.1-cp311-cp311-linux_x86_64.whl size=1213478 sha256=a7e4cf3a4c3fb459b3e4046f57327bce211ceedcb281672730b44a6a4e6fed9c\r\n",
      "  Stored in directory: /root/.cache/pip/wheels/d5/f0/56/54ed274b63aa20da572c894203e0060cd552c6fe01759043f5\r\n",
      "Successfully built pillow\r\n",
      "Installing collected packages: pillow\r\n",
      "  Attempting uninstall: pillow\r\n",
      "    Found existing installation: pillow 11.2.1\r\n",
      "    Uninstalling pillow-11.2.1:\r\n",
      "      Successfully uninstalled pillow-11.2.1\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery-storage<3.0.0,>=2.30.0, which is not installed.\r\n",
      "torchtune 0.6.1 requires Pillow>=9.4.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "imageio 2.37.0 requires pillow>=8.3.2, but you have pillow 8.3.1 which is incompatible.\r\n",
      "dopamine-rl 4.1.2 requires gymnasium>=1.0.0, but you have gymnasium 0.29.0 which is incompatible.\r\n",
      "scikit-image 0.25.2 requires pillow>=10.1, but you have pillow 8.3.1 which is incompatible.\r\n",
      "torchvision 0.21.0+cu124 requires pillow!=8.3.*,>=5.3.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "fastai 2.7.19 requires pillow>=9.0.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "plotnine 0.14.5 requires matplotlib>=3.8.0, but you have matplotlib 3.7.2 which is incompatible.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery[bqstorage,pandas]>=3.31.0, but you have google-cloud-bigquery 3.25.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires rich<14,>=12.4.4, but you have rich 14.0.0 which is incompatible.\r\n",
      "mlxtend 0.23.4 requires scikit-learn>=1.3.1, but you have scikit-learn 1.2.2 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed pillow-8.3.1\r\n",
      "\u001b[31mERROR: Ignored the following yanked versions: 1.11.0, 1.14.0rc1\u001b[0m\u001b[31m\r\n",
      "\u001b[0m\u001b[31mERROR: Ignored the following versions that require a different python version: 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10; 1.7.2 Requires-Python >=3.7,<3.11; 1.7.3 Requires-Python >=3.7,<3.11; 1.8.0 Requires-Python >=3.8,<3.11; 1.8.0rc1 Requires-Python >=3.8,<3.11; 1.8.0rc2 Requires-Python >=3.8,<3.11; 1.8.0rc3 Requires-Python >=3.8,<3.11; 1.8.0rc4 Requires-Python >=3.8,<3.11; 1.8.1 Requires-Python >=3.8,<3.11\u001b[0m\u001b[31m\r\n",
      "\u001b[0m\u001b[31mERROR: Could not find a version that satisfies the requirement scipy==1.7.1 (from versions: 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0, 0.19.1, 1.0.0, 1.0.1, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0, 1.4.1, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.9.0rc1, 1.9.0rc2, 1.9.0rc3, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0rc1, 1.10.0rc2, 1.10.0, 1.10.1, 1.11.0rc1, 1.11.0rc2, 1.11.1, 1.11.2, 1.11.3, 1.11.4, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.13.0rc1, 1.13.0, 1.13.1, 1.14.0rc2, 1.14.0, 1.14.1, 1.15.0rc1, 1.15.0rc2, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.16.0rc1, 1.16.0rc2, 1.16.0, 1.16.1, 1.16.2)\u001b[0m\u001b[31m\r\n",
      "\u001b[0m\u001b[31mERROR: No matching distribution found for scipy==1.7.1\u001b[0m\u001b[31m\r\n",
      "\u001b[0mCollecting requests==2.26.0\r\n",
      "  Downloading requests-2.26.0-py2.py3-none-any.whl.metadata (4.8 kB)\r\n",
      "Collecting urllib3<1.27,>=1.21.1 (from requests==2.26.0)\r\n",
      "  Downloading urllib3-1.26.20-py2.py3-none-any.whl.metadata (50 kB)\r\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m50.1/50.1 kB\u001b[0m \u001b[31m1.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hRequirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.11/dist-packages (from requests==2.26.0) (2025.6.15)\r\n",
      "Collecting charset-normalizer~=2.0.0 (from requests==2.26.0)\r\n",
      "  Downloading charset_normalizer-2.0.12-py3-none-any.whl.metadata (11 kB)\r\n",
      "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.11/dist-packages (from requests==2.26.0) (3.10)\r\n",
      "Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m62.3/62.3 kB\u001b[0m \u001b[31m2.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading charset_normalizer-2.0.12-py3-none-any.whl (39 kB)\r\n",
      "Downloading urllib3-1.26.20-py2.py3-none-any.whl (144 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m144.2/144.2 kB\u001b[0m \u001b[31m4.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: urllib3, charset-normalizer, requests\r\n",
      "  Attempting uninstall: urllib3\r\n",
      "    Found existing installation: urllib3 2.5.0\r\n",
      "    Uninstalling urllib3-2.5.0:\r\n",
      "      Successfully uninstalled urllib3-2.5.0\r\n",
      "  Attempting uninstall: charset-normalizer\r\n",
      "    Found existing installation: charset-normalizer 3.4.2\r\n",
      "    Uninstalling charset-normalizer-3.4.2:\r\n",
      "      Successfully uninstalled charset-normalizer-3.4.2\r\n",
      "  Attempting uninstall: requests\r\n",
      "    Found existing installation: requests 2.32.4\r\n",
      "    Uninstalling requests-2.32.4:\r\n",
      "      Successfully uninstalled requests-2.32.4\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery-storage<3.0.0,>=2.30.0, which is not installed.\r\n",
      "libpysal 4.9.2 requires requests>=2.27, but you have requests 2.26.0 which is incompatible.\r\n",
      "datasets 3.6.0 requires fsspec[http]<=2025.3.0,>=2023.1.0, but you have fsspec 2025.5.1 which is incompatible.\r\n",
      "datasets 3.6.0 requires requests>=2.32.2, but you have requests 2.26.0 which is incompatible.\r\n",
      "jupyterlab-server 2.27.3 requires requests>=2.31, but you have requests 2.26.0 which is incompatible.\r\n",
      "google-colab 1.0.0 requires google-auth==2.38.0, but you have google-auth 2.40.3 which is incompatible.\r\n",
      "google-colab 1.0.0 requires notebook==6.5.7, but you have notebook 6.5.4 which is incompatible.\r\n",
      "google-colab 1.0.0 requires pandas==2.2.2, but you have pandas 2.2.3 which is incompatible.\r\n",
      "google-colab 1.0.0 requires requests==2.32.3, but you have requests 2.26.0 which is incompatible.\r\n",
      "google-colab 1.0.0 requires tornado==6.4.2, but you have tornado 6.5.1 which is incompatible.\r\n",
      "pandas-gbq 0.29.1 requires google-api-core<3.0.0,>=2.10.2, but you have google-api-core 1.34.1 which is incompatible.\r\n",
      "google-genai 1.21.1 requires requests<3.0.0,>=2.28.1, but you have requests 2.26.0 which is incompatible.\r\n",
      "gcsfs 2025.3.2 requires fsspec==2025.3.2, but you have fsspec 2025.5.1 which is incompatible.\r\n",
      "langchain-core 0.3.66 requires packaging<25,>=23.2, but you have packaging 25.0 which is incompatible.\r\n",
      "fastai 2.7.19 requires pillow>=9.0.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "google-cloud-storage 2.19.0 requires google-api-core<3.0.0dev,>=2.15.0, but you have google-api-core 1.34.1 which is incompatible.\r\n",
      "yfinance 0.2.63 requires requests>=2.31, but you have requests 2.26.0 which is incompatible.\r\n",
      "dataproc-spark-connect 0.7.5 requires google-api-core>=2.19, but you have google-api-core 1.34.1 which is incompatible.\r\n",
      "sphinx 8.2.3 requires requests>=2.30.0, but you have requests 2.26.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery[bqstorage,pandas]>=3.31.0, but you have google-cloud-bigquery 3.25.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires requests>=2.27.1, but you have requests 2.26.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires rich<14,>=12.4.4, but you have rich 14.0.0 which is incompatible.\r\n",
      "jupyter-kernel-gateway 2.5.2 requires jupyter-client<8.0,>=5.2.0, but you have jupyter-client 8.6.3 which is incompatible.\r\n",
      "tweepy 4.15.0 requires requests<3,>=2.27.0, but you have requests 2.26.0 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed charset-normalizer-2.0.12 requests-2.26.0 urllib3-1.26.20\r\n",
      "Collecting tqdm==4.62.2\r\n",
      "  Downloading tqdm-4.62.2-py2.py3-none-any.whl.metadata (56 kB)\r\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m57.0/57.0 kB\u001b[0m \u001b[31m1.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hDownloading tqdm-4.62.2-py2.py3-none-any.whl (76 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m76.2/76.2 kB\u001b[0m \u001b[31m2.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: tqdm\r\n",
      "  Attempting uninstall: tqdm\r\n",
      "    Found existing installation: tqdm 4.67.1\r\n",
      "    Uninstalling tqdm-4.67.1:\r\n",
      "      Successfully uninstalled tqdm-4.67.1\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "torchtune 0.6.1 requires Pillow>=9.4.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "datasets 3.6.0 requires fsspec[http]<=2025.3.0,>=2023.1.0, but you have fsspec 2025.5.1 which is incompatible.\r\n",
      "datasets 3.6.0 requires requests>=2.32.2, but you have requests 2.26.0 which is incompatible.\r\n",
      "datasets 3.6.0 requires tqdm>=4.66.3, but you have tqdm 4.62.2 which is incompatible.\r\n",
      "featuretools 1.31.0 requires tqdm>=4.66.3, but you have tqdm 4.62.2 which is incompatible.\r\n",
      "preprocessing 0.1.13 requires nltk==3.2.4, but you have nltk 3.9.1 which is incompatible.\r\n",
      "dopamine-rl 4.1.2 requires gymnasium>=1.0.0, but you have gymnasium 0.29.0 which is incompatible.\r\n",
      "dopamine-rl 4.1.2 requires tqdm>=4.64.1, but you have tqdm 4.62.2 which is incompatible.\r\n",
      "dataproc-spark-connect 0.7.5 requires google-api-core>=2.19, but you have google-api-core 1.34.1 which is incompatible.\r\n",
      "dataproc-spark-connect 0.7.5 requires tqdm>=4.67, but you have tqdm 4.62.2 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed tqdm-4.62.2\r\n",
      "Collecting ninja==1.10.2\r\n",
      "  Downloading ninja-1.10.2-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl.metadata (5.0 kB)\r\n",
      "Downloading ninja-1.10.2-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (108 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m108.1/108.1 kB\u001b[0m \u001b[31m2.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: ninja\r\n",
      "  Attempting uninstall: ninja\r\n",
      "    Found existing installation: ninja 1.11.1.4\r\n",
      "    Uninstalling ninja-1.11.1.4:\r\n",
      "      Successfully uninstalled ninja-1.11.1.4\r\n",
      "Successfully installed ninja-1.10.2\r\n",
      "Collecting matplotlib==3.4.2\r\n",
      "  Downloading matplotlib-3.4.2.tar.gz (37.3 MB)\r\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m37.3/37.3 MB\u001b[0m \u001b[31m40.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\r\n",
      "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (0.12.1)\r\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (1.4.8)\r\n",
      "Requirement already satisfied: numpy>=1.16 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (1.26.4)\r\n",
      "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (8.3.1)\r\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (3.0.9)\r\n",
      "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.11/dist-packages (from matplotlib==3.4.2) (2.9.0.post0)\r\n",
      "Requirement already satisfied: mkl_fft in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (1.3.8)\r\n",
      "Requirement already satisfied: mkl_random in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (1.2.4)\r\n",
      "Requirement already satisfied: mkl_umath in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (0.1.1)\r\n",
      "Requirement already satisfied: mkl in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (2025.2.0)\r\n",
      "Requirement already satisfied: tbb4py in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (2022.2.0)\r\n",
      "Requirement already satisfied: mkl-service in /usr/local/lib/python3.11/dist-packages (from numpy>=1.16->matplotlib==3.4.2) (2.4.1)\r\n",
      "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.11/dist-packages (from python-dateutil>=2.7->matplotlib==3.4.2) (1.17.0)\r\n",
      "Requirement already satisfied: intel-openmp<2026,>=2024 in /usr/local/lib/python3.11/dist-packages (from mkl->numpy>=1.16->matplotlib==3.4.2) (2024.2.0)\r\n",
      "Requirement already satisfied: tbb==2022.* in /usr/local/lib/python3.11/dist-packages (from mkl->numpy>=1.16->matplotlib==3.4.2) (2022.2.0)\r\n",
      "Requirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.11/dist-packages (from tbb==2022.*->mkl->numpy>=1.16->matplotlib==3.4.2) (1.4.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.11/dist-packages (from mkl_umath->numpy>=1.16->matplotlib==3.4.2) (2024.2.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.11/dist-packages (from intel-openmp<2026,>=2024->mkl->numpy>=1.16->matplotlib==3.4.2) (2024.2.0)\r\n",
      "Building wheels for collected packages: matplotlib\r\n",
      "  Building wheel for matplotlib (setup.py) ... \u001b[?25l\u001b[?25hdone\r\n",
      "  Created wheel for matplotlib: filename=matplotlib-3.4.2-cp311-cp311-linux_x86_64.whl size=10440320 sha256=50325636952dcb3c741115bf08b5501f123b06c05a7583e1a07bd63bba7b1d37\r\n",
      "  Stored in directory: /root/.cache/pip/wheels/57/a1/ed/118b1370d20bb050979204b93d5822952a2b34e36635d27908\r\n",
      "Successfully built matplotlib\r\n",
      "Installing collected packages: matplotlib\r\n",
      "  Attempting uninstall: matplotlib\r\n",
      "    Found existing installation: matplotlib 3.7.2\r\n",
      "    Uninstalling matplotlib-3.7.2:\r\n",
      "      Successfully uninstalled matplotlib-3.7.2\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery-storage<3.0.0,>=2.30.0, which is not installed.\r\n",
      "ydata-profiling 4.16.1 requires matplotlib<=3.10,>=3.5, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "cartopy 0.24.1 requires matplotlib>=3.6, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "ipympl 0.9.7 requires matplotlib<4,>=3.5.0, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "mne 1.9.0 requires matplotlib>=3.6, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "arviz 0.21.0 requires matplotlib>=3.5, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "fastai 2.7.19 requires pillow>=9.0.0, but you have pillow 8.3.1 which is incompatible.\r\n",
      "plotnine 0.14.5 requires matplotlib>=3.8.0, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "bigframes 2.8.0 requires google-cloud-bigquery[bqstorage,pandas]>=3.31.0, but you have google-cloud-bigquery 3.25.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires matplotlib>=3.7.1, but you have matplotlib 3.4.2 which is incompatible.\r\n",
      "bigframes 2.8.0 requires requests>=2.27.1, but you have requests 2.26.0 which is incompatible.\r\n",
      "bigframes 2.8.0 requires rich<14,>=12.4.4, but you have rich 14.0.0 which is incompatible.\r\n",
      "mlxtend 0.23.4 requires scikit-learn>=1.3.1, but you have scikit-learn 1.2.2 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed matplotlib-3.4.2\r\n",
      "Collecting imageio==2.9.0\r\n",
      "  Downloading imageio-2.9.0-py3-none-any.whl.metadata (2.6 kB)\r\n",
      "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from imageio==2.9.0) (1.26.4)\r\n",
      "Requirement already satisfied: pillow in /usr/local/lib/python3.11/dist-packages (from imageio==2.9.0) (8.3.1)\r\n",
      "Requirement already satisfied: mkl_fft in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (1.3.8)\r\n",
      "Requirement already satisfied: mkl_random in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (1.2.4)\r\n",
      "Requirement already satisfied: mkl_umath in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (0.1.1)\r\n",
      "Requirement already satisfied: mkl in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (2025.2.0)\r\n",
      "Requirement already satisfied: tbb4py in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (2022.2.0)\r\n",
      "Requirement already satisfied: mkl-service in /usr/local/lib/python3.11/dist-packages (from numpy->imageio==2.9.0) (2.4.1)\r\n",
      "Requirement already satisfied: intel-openmp<2026,>=2024 in /usr/local/lib/python3.11/dist-packages (from mkl->numpy->imageio==2.9.0) (2024.2.0)\r\n",
      "Requirement already satisfied: tbb==2022.* in /usr/local/lib/python3.11/dist-packages (from mkl->numpy->imageio==2.9.0) (2022.2.0)\r\n",
      "Requirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.11/dist-packages (from tbb==2022.*->mkl->numpy->imageio==2.9.0) (1.4.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.11/dist-packages (from mkl_umath->numpy->imageio==2.9.0) (2024.2.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.11/dist-packages (from intel-openmp<2026,>=2024->mkl->numpy->imageio==2.9.0) (2024.2.0)\r\n",
      "Downloading imageio-2.9.0-py3-none-any.whl (3.3 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.3/3.3 MB\u001b[0m \u001b[31m36.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: imageio\r\n",
      "  Attempting uninstall: imageio\r\n",
      "    Found existing installation: imageio 2.37.0\r\n",
      "    Uninstalling imageio-2.37.0:\r\n",
      "      Successfully uninstalled imageio-2.37.0\r\n",
      "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\r\n",
      "scikit-image 0.25.2 requires imageio!=2.35.0,>=2.33, but you have imageio 2.9.0 which is incompatible.\r\n",
      "scikit-image 0.25.2 requires pillow>=10.1, but you have pillow 8.3.1 which is incompatible.\u001b[0m\u001b[31m\r\n",
      "\u001b[0mSuccessfully installed imageio-2.9.0\r\n",
      "Collecting imgui==1.3.0\r\n",
      "  Downloading imgui-1.3.0.tar.gz (844 kB)\r\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m844.7/844.7 kB\u001b[0m \u001b[31m15.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25h  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\r\n",
      "  \r\n",
      "  \u001b[31m×\u001b[0m \u001b[32mpython setup.py egg_info\u001b[0m did not run successfully.\r\n",
      "  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\r\n",
      "  \u001b[31m╰─>\u001b[0m See above for output.\r\n",
      "  \r\n",
      "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\r\n",
      "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25herror\r\n",
      "\u001b[1;31merror\u001b[0m: \u001b[1mmetadata-generation-failed\u001b[0m\r\n",
      "\r\n",
      "\u001b[31m×\u001b[0m Encountered error while generating package metadata.\r\n",
      "\u001b[31m╰─>\u001b[0m See above for output.\r\n",
      "\r\n",
      "\u001b[1;35mnote\u001b[0m: This is an issue with the package mentioned above, not pip.\r\n",
      "\u001b[1;36mhint\u001b[0m: See above for details.\r\n",
      "Collecting glfw==2.2.0\r\n",
      "  Downloading glfw-2.2.0-py2.py27.py3.py30.py31.py32.py33.py34.py35.py36.py37.py38-none-manylinux2014_x86_64.whl.metadata (4.3 kB)\r\n",
      "Downloading glfw-2.2.0-py2.py27.py3.py30.py31.py32.py33.py34.py35.py36.py37.py38-none-manylinux2014_x86_64.whl (204 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m204.8/204.8 kB\u001b[0m \u001b[31m5.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: glfw\r\n",
      "Successfully installed glfw-2.2.0\r\n",
      "Collecting pyopengl==3.1.5\r\n",
      "  Downloading PyOpenGL-3.1.5-py3-none-any.whl.metadata (3.2 kB)\r\n",
      "Downloading PyOpenGL-3.1.5-py3-none-any.whl (2.4 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.4/2.4 MB\u001b[0m \u001b[31m29.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: pyopengl\r\n",
      "  Attempting uninstall: pyopengl\r\n",
      "    Found existing installation: PyOpenGL 3.1.9\r\n",
      "    Uninstalling PyOpenGL-3.1.9:\r\n",
      "      Successfully uninstalled PyOpenGL-3.1.9\r\n",
      "Successfully installed pyopengl-3.1.5\r\n",
      "Collecting imageio-ffmpeg==0.4.3\r\n",
      "  Downloading imageio_ffmpeg-0.4.3-py3-none-manylinux2010_x86_64.whl.metadata (1.6 kB)\r\n",
      "Downloading imageio_ffmpeg-0.4.3-py3-none-manylinux2010_x86_64.whl (26.9 MB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m26.9/26.9 MB\u001b[0m \u001b[31m52.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: imageio-ffmpeg\r\n",
      "  Attempting uninstall: imageio-ffmpeg\r\n",
      "    Found existing installation: imageio-ffmpeg 0.6.0\r\n",
      "    Uninstalling imageio-ffmpeg-0.6.0:\r\n",
      "      Successfully uninstalled imageio-ffmpeg-0.6.0\r\n",
      "Successfully installed imageio-ffmpeg-0.4.3\r\n",
      "Collecting pyspng\r\n",
      "  Downloading pyspng-0.1.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.0 kB)\r\n",
      "Requirement already satisfied: numpy in /usr/local/lib/python3.11/dist-packages (from pyspng) (1.26.4)\r\n",
      "Requirement already satisfied: mkl_fft in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (1.3.8)\r\n",
      "Requirement already satisfied: mkl_random in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (1.2.4)\r\n",
      "Requirement already satisfied: mkl_umath in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (0.1.1)\r\n",
      "Requirement already satisfied: mkl in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (2025.2.0)\r\n",
      "Requirement already satisfied: tbb4py in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (2022.2.0)\r\n",
      "Requirement already satisfied: mkl-service in /usr/local/lib/python3.11/dist-packages (from numpy->pyspng) (2.4.1)\r\n",
      "Requirement already satisfied: intel-openmp<2026,>=2024 in /usr/local/lib/python3.11/dist-packages (from mkl->numpy->pyspng) (2024.2.0)\r\n",
      "Requirement already satisfied: tbb==2022.* in /usr/local/lib/python3.11/dist-packages (from mkl->numpy->pyspng) (2022.2.0)\r\n",
      "Requirement already satisfied: tcmlib==1.* in /usr/local/lib/python3.11/dist-packages (from tbb==2022.*->mkl->numpy->pyspng) (1.4.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-rt in /usr/local/lib/python3.11/dist-packages (from mkl_umath->numpy->pyspng) (2024.2.0)\r\n",
      "Requirement already satisfied: intel-cmplr-lib-ur==2024.2.0 in /usr/local/lib/python3.11/dist-packages (from intel-openmp<2026,>=2024->mkl->numpy->pyspng) (2024.2.0)\r\n",
      "Downloading pyspng-0.1.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (196 kB)\r\n",
      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m196.1/196.1 kB\u001b[0m \u001b[31m3.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\r\n",
      "\u001b[?25hInstalling collected packages: pyspng\r\n",
      "Successfully installed pyspng-0.1.3\r\n"
     ]
    }
   ],
   "source": [
    "# 根据environment.yml安装主要的依赖包（注意版本以兼容 Kaggle 环境）\n",
    "!pip install numpy>=1.20\n",
    "!pip install click>=8.0\n",
    "!pip install pillow==8.3.1\n",
    "!pip install scipy==1.7.1\n",
    "!pip install requests==2.26.0\n",
    "!pip install tqdm==4.62.2\n",
    "!pip install ninja==1.10.2\n",
    "!pip install matplotlib==3.4.2\n",
    "!pip install imageio==2.9.0\n",
    "\n",
    "!pip install imgui==1.3.0\n",
    "!pip install glfw==2.2.0\n",
    "!pip install pyopengl==3.1.5\n",
    "!pip install imageio-ffmpeg==0.4.3\n",
    "!pip install pyspng\n",
    "\n",
    "# PyTorch 在 Kaggle 中已经预装，通常不需要重新安装\n",
    "# 如果需要特定版本，可以这样安装：\n",
    "# !pip install torch==1.9.1+cu111 -f https://download.pytorch.org/whl/torch_stable.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "781e1949",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:26:02.978905Z",
     "iopub.status.busy": "2025-09-27T23:26:02.978505Z",
     "iopub.status.idle": "2025-09-27T23:26:09.584277Z",
     "shell.execute_reply": "2025-09-27T23:26:09.582573Z"
    },
    "papermill": {
     "duration": 6.619617,
     "end_time": "2025-09-27T23:26:09.586222",
     "exception": false,
     "start_time": "2025-09-27T23:26:02.966605",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PyTorch版本: 2.6.0+cu124\n",
      "CUDA available: False\n",
      "Pillow版本: 11.2.1\n",
      "SciPy版本: 1.15.3\n"
     ]
    }
   ],
   "source": [
    "import torch\n",
    "print(\"PyTorch版本:\", torch.__version__)\n",
    "print(\"CUDA available:\", torch.cuda.is_available())\n",
    "\n",
    "import PIL\n",
    "print(\"Pillow版本:\", PIL.__version__)\n",
    "\n",
    "import scipy\n",
    "print(\"SciPy版本:\", scipy.__version__)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f21882db",
   "metadata": {
    "papermill": {
     "duration": 0.009737,
     "end_time": "2025-09-27T23:26:09.606679",
     "exception": false,
     "start_time": "2025-09-27T23:26:09.596942",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "#以下为训练部分# "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ffbb7fd",
   "metadata": {
    "papermill": {
     "duration": 0.011055,
     "end_time": "2025-09-27T23:26:09.627845",
     "exception": false,
     "start_time": "2025-09-27T23:26:09.616790",
     "status": "completed"
    },
    "tags": []
   },
   "source": [
    "👇此code块是用dataset_tool.py将数据打包"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "989bdf6a",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:26:09.650659Z",
     "iopub.status.busy": "2025-09-27T23:26:09.650153Z",
     "iopub.status.idle": "2025-09-27T23:26:15.489417Z",
     "shell.execute_reply": "2025-09-27T23:26:15.488204Z"
    },
    "papermill": {
     "duration": 5.853274,
     "end_time": "2025-09-27T23:26:15.491332",
     "exception": false,
     "start_time": "2025-09-27T23:26:09.638058",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "转换图像格式: 100%|██████████| 42/42 [00:04<00:00,  9.98it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "打包命令输出：\n",
      "\n",
      "错误信息：\n",
      "\n",
      "  0%|          | 0/42 [00:00<?, ?it/s]\n",
      " 10%|▉         | 4/42 [00:00<00:01, 35.04it/s]\n",
      " 19%|█▉        | 8/42 [00:00<00:01, 25.93it/s]\n",
      " 33%|███▎      | 14/42 [00:00<00:00, 35.28it/s]\n",
      " 45%|████▌     | 19/42 [00:00<00:00, 38.74it/s]\n",
      " 57%|█████▋    | 24/42 [00:00<00:00, 40.21it/s]\n",
      " 69%|██████▉   | 29/42 [00:00<00:00, 39.91it/s]\n",
      " 81%|████████  | 34/42 [00:00<00:00, 39.53it/s]\n",
      " 93%|█████████▎| 39/42 [00:01<00:00, 40.57it/s]\n",
      "100%|██████████| 42/42 [00:01<00:00, 38.94it/s]\n",
      "\n",
      "\n",
      "数据集打包成功，保存至：/kaggle/working/my_dataset-512x512.zip\n"
     ]
    }
   ],
   "source": [
    "import subprocess\n",
    "import os\n",
    "from PIL import Image\n",
    "from tqdm import tqdm\n",
    "\n",
    "\n",
    "# 先将图片自动处理为RGB模式（默认为RGBA）\n",
    "# 1. 定义路径\n",
    "input_dir = \"/kaggle/input/ruishou/ruishou\"  # 原始图像目录\n",
    "temp_rgb_dir = \"/kaggle/working/temp_rgb\"  # 临时存放RGB转换后的图像\n",
    "output_zip = \"/kaggle/working/my_dataset-512x512.zip\"  # 最终输出的ZIP文件\n",
    "\n",
    "# 2. 创建临时目录\n",
    "os.makedirs(temp_rgb_dir, exist_ok=True)\n",
    "\n",
    "# 3. 批量转换图像为RGB格式\n",
    "image_files = [f for f in os.listdir(input_dir) if f.lower().endswith(('.png', '.jpg', '.jpeg'))]\n",
    "\n",
    "for filename in tqdm(image_files, desc=\"转换图像格式\"):\n",
    "    try:\n",
    "        # 打开图像\n",
    "        img_path = os.path.join(input_dir, filename)\n",
    "        with Image.open(img_path) as img:\n",
    "            # 转换为RGB（自动处理RGBA、P模式等）\n",
    "            if img.mode not in ('RGB', 'L'):  # 'L'是灰度模式\n",
    "                img = img.convert('RGB')\n",
    "            # 保存到临时目录\n",
    "            save_path = os.path.join(temp_rgb_dir, filename)\n",
    "            img.save(save_path)\n",
    "    except Exception as e:\n",
    "        print(f\"处理 {filename} 失败: {e}\")\n",
    "\n",
    "\n",
    "\n",
    "# 4. 使用转换后的RGB图像进行打包\n",
    "command = [\n",
    "    \"python\", \"dataset_tool.py\",\n",
    "    \"--source\", temp_rgb_dir,\n",
    "    \"--dest\", output_zip,\n",
    "    \"--resolution\", \"512x512\"\n",
    "]\n",
    "\n",
    "result = subprocess.run(\n",
    "    command,\n",
    "    capture_output=True,\n",
    "    text=True\n",
    ")\n",
    "\n",
    "# 打印结果\n",
    "print(\"\\n打包命令输出：\")\n",
    "print(result.stdout)\n",
    "print(\"错误信息：\")\n",
    "print(result.stderr)\n",
    "\n",
    "if result.returncode == 0:\n",
    "    print(f\"\\n数据集打包成功，保存至：{output_zip}\")\n",
    "else:\n",
    "    print(\"\\n数据集打包失败\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a26369d1",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:26:15.525394Z",
     "iopub.status.busy": "2025-09-27T23:26:15.525041Z",
     "iopub.status.idle": "2025-09-27T23:26:15.544298Z",
     "shell.execute_reply": "2025-09-27T23:26:15.542884Z"
    },
    "papermill": {
     "duration": 0.034447,
     "end_time": "2025-09-27T23:26:15.546065",
     "exception": false,
     "start_time": "2025-09-27T23:26:15.511618",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "文件已替换: /kaggle/working/stylegan3/train.py\n",
      "替换后的文件大小: 15914 字节\n"
     ]
    }
   ],
   "source": [
    "#替换train.py为正确版本\n",
    "\n",
    "import shutil\n",
    "import os\n",
    "\n",
    "\n",
    "source_path = \"/kaggle/input/real-train-py/train.py\"  \n",
    "destination_path = \"/kaggle/working/stylegan3/train.py\"\n",
    "\n",
    "# 确保目标目录存在\n",
    "os.makedirs(os.path.dirname(destination_path), exist_ok=True)\n",
    "\n",
    "# 复制并替换文件\n",
    "shutil.copy2(source_path, destination_path)\n",
    "\n",
    "# 验证替换是否成功\n",
    "print(f\"文件已替换: {destination_path}\")\n",
    "print(f\"替换后的文件大小: {os.path.getsize(destination_path)} 字节\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e17029ab",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:26:15.570960Z",
     "iopub.status.busy": "2025-09-27T23:26:15.570412Z",
     "iopub.status.idle": "2025-09-27T23:26:15.706959Z",
     "shell.execute_reply": "2025-09-27T23:26:15.705271Z"
    },
    "papermill": {
     "duration": 0.151581,
     "end_time": "2025-09-27T23:26:15.709197",
     "exception": false,
     "start_time": "2025-09-27T23:26:15.557616",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: /kaggle/working/training/log.txt: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "# 用2个GPU训练StyleGAN3-T\n",
    "\n",
    "# --outdir：训练结果输出目录\n",
    "# --cfg：模型配置（此处为stylegan3-t）\n",
    "# --data：数据集路径\n",
    "# --gpus：使用的GPU数量（Kaggle最大限制为2）\n",
    "# --batch：总批次大小（所有GPU合计）\n",
    "# --batch-gpu：每个GPU的子批次大小\n",
    "# --gamma：训练超参数（影响生成质量的平衡参数）\n",
    "# --mirror：是否启用数据镜像增强（1=启用，0=禁用）\n",
    "# --kimg：总训练图像数量（单位：千张）\n",
    "# --snap：快照保存间隔（单位：千张图像）\n",
    "!python train.py \\\n",
    "--outdir /kaggle/working/training \\\n",
    "--cfg stylegan3-t \\\n",
    "--data /kaggle/working/my_dataset-512x512.zip \\\n",
    "--gpus 2 \\\n",
    "--batch 8 \\\n",
    "--batch-gpu 4 \\\n",
    "--gamma 8.2 \\\n",
    "--aug ada \\\n",
    "--resume /kaggle/input/100kimg-pkl/network-snapshot-000100.pkl \\\n",
    "--mirror 1 \\\n",
    "--kimg 100 \\\n",
    "--snap 5 \\\n",
    "> /kaggle/working/training/log.txt 2>&1\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "20533f5f",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2025-09-27T23:26:15.734215Z",
     "iopub.status.busy": "2025-09-27T23:26:15.733831Z",
     "iopub.status.idle": "2025-09-27T23:26:15.870463Z",
     "shell.execute_reply": "2025-09-27T23:26:15.869157Z"
    },
    "papermill": {
     "duration": 0.152549,
     "end_time": "2025-09-27T23:26:15.873286",
     "exception": false,
     "start_time": "2025-09-27T23:26:15.720737",
     "status": "completed"
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\tzip warning: name not matched: /kaggle/working/training\r\n",
      "\r\n",
      "zip error: Nothing to do! (try: zip -r /kaggle/working/training.zip . -i /kaggle/working/training)\r\n"
     ]
    }
   ],
   "source": [
    "!zip -r /kaggle/working/training.zip /kaggle/working/training"
   ]
  }
 ],
 "metadata": {
  "kaggle": {
   "accelerator": "none",
   "dataSources": [
    {
     "datasetId": 8350728,
     "sourceId": 13177960,
     "sourceType": "datasetVersion"
    },
    {
     "datasetId": 8352319,
     "sourceId": 13180091,
     "sourceType": "datasetVersion"
    },
    {
     "datasetId": 8358424,
     "sourceId": 13189516,
     "sourceType": "datasetVersion"
    }
   ],
   "dockerImageVersionId": 31090,
   "isGpuEnabled": false,
   "isInternetEnabled": true,
   "language": "python",
   "sourceType": "notebook"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.13"
  },
  "papermill": {
   "default_parameters": {},
   "duration": 293.078612,
   "end_time": "2025-09-27T23:26:18.755478",
   "environment_variables": {},
   "exception": null,
   "input_path": "__notebook__.ipynb",
   "output_path": "__notebook__.ipynb",
   "parameters": {},
   "start_time": "2025-09-27T23:21:25.676866",
   "version": "2.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
